examples/hyper.rs

```diff
@@ -72,16 +76,70 @@ async fn main() {
         .insert("/blog", BoxCloneService::new(service_fn(blog)).into())
         .unwrap();
 
+    let listener = TcpListener::bind(("127.0.0.1", 3000)).await.unwrap();
+
     // boilerplate for the hyper service
     let router = Arc::new(router);
-    let make_service = make_service_fn(|_| {
+
+    loop {
         let router = router.clone();
-        async { Ok::<_, Infallible>(service_fn(move |request| route(router.clone(), request))) }
-    });
-
-    // run the server
-    Server::bind(&([127, 0, 0, 1], 3000).into())
-        .serve(make_service)
-        .await
-        .unwrap()
+        let (tcp, _) = listener.accept().await.unwrap();
+        tokio::task::spawn(async move {
+            if let Err(err) = ConnectionBuilder::new()
+                .serve_connection(
+                    TokioIo::new(tcp),
+                    hyper::service::service_fn(|request| async {
+                        route(router.clone(), request).await
+                    }),
+                )
+                .await
+            {
+                println!("Error serving connection: {:?}", err);
+            }
+        });
+    }
+}
+
+mod body {
+    use std::convert::Infallible;
+    use std::pin::Pin;
+    use std::task::{Context, Poll};
+
+    use hyper::body::{Body as HttpBody, Bytes, Frame};
+
+    pub enum Body {
```

Comment:
Can we use [`hyper_body_util::Full<Bytes>`](https://docs.rs/http-body-util/0.1.2/http_body_util/struct.Full.html) for the body type instead? This code is a little distracting from the main router code.

Reply:
> Can we use [hyper_body_util::Full<Bytes>](https://docs.rs/http-body-util/0.1.2/http_body_util/struct.Full.html) for the body type instead?

Yes. The easy way is just using `http_body_util::Full<Bytes>`, to describe empty body, using `Full::default` constructor.

Reply:
Updated.